coordinator/agent/agent.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package agent
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-load-test-ng/api"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest"
+)
+
+// LoadAgent is the object acting as a client to the load-test agent
+// HTTP API.
+type LoadAgent struct {
+	config LoadAgentConfig
+	status loadtest.Status
+	client *http.Client
+}
+
+// New creates and initializes a new LoadAgent for the given config.
+// An error is returned if the initialization fails.
+func New(config LoadAgentConfig) (*LoadAgent, error) {
+	if ok, err := config.IsValid(); !ok {
+		return nil, err
+	}
+	return &LoadAgent{
+		config: config,
+		status: loadtest.Status{},
+		client: &http.Client{},
+	}, nil
+}
+
+func (a *LoadAgent) apiRequest(req *http.Request) error {
+	resp, err := a.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("agent: bad response status code %d", resp.StatusCode)
+	}
+	res := &api.Response{}
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return err
+	}
+	if res.Error != "" {
+		return fmt.Errorf("agent: api request error: %s", res.Error)
+	}
+	a.status = res.Status
+	return nil
+}
+
+func (a *LoadAgent) AddUsers(n int) error {
+	url := fmt.Sprintf("%s/loadagent/%s/addusers?amount=%d", a.config.ApiURL, a.config.Id, n)
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	if err := a.apiRequest(req); err != nil {
```

Comment:
why not just return 'err' ?

Reply:
I guess it was me being extra cautious since I've been burnt by nil error interfaces before. In this case case though it looks like it's perfectly fine to return directly the result of `apiRequest`.